genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/SetUpJobAction.java

```diff
@@ -513,7 +514,7 @@ private File createJobEnvironmentFile(
         // Variables in environment file are base64 encoded to avoid escaping, quoting.
         // Decode all values.
         env.keySet().forEach(key ->
-            env.compute(key, (k, v) -> new String(Base64.decodeBase64(v)))
+            env.compute(key, (k, v) -> new String(Base64.decodeBase64(v), StandardCharsets.UTF_8))
```

Comment:
@mprimi `Spotbugs` was complaining about this not being portable due to internationalization rules since we didn't provide a character encoding. I added `UTF-8` but wanted to check with you if you think this is ok. I looked at the bash script where you're encoding the variables and didn't see anything specific to encoding.

Reply:
Refreshing this now that the commit exists again. @mprimi take a look

Reply:
Encoding would transform everything into plain old ASCII, so parsing as UTF-8 should be fine.